src/background/backgroundHelpers.ts

```diff
@@ -0,0 +1,27 @@
+import { IBytesRepository } from "../data/bytes/IBytesRepository";
+
+export const addBytesTransferred = async (bytes: number) => {
+    IBytesRepository.instance.addBytesTransferred(bytes);
+
+    try {
+        await chrome.runtime.sendMessage({
+            command: {
+                bytesTransferredChanged:
+                    IBytesRepository.instance.getBytesTransferred(),
+            },
+        });
+    } catch (e: unknown) {
+        if (e instanceof Error) {
+            if (
+                e.message ===
+                "Could not establish connection. Receiving end does not exist."
+            ) {
+                console.warn(
+                    `Error Caught: ${e}\nIf popup is open and this error is seen in the console, debugging is required.`
```

Comment:
Debugging is required as soon as you have an error, isn't it? We should maybe consider it the other way around "if popup is closed then the error is ok"

Reply:
I can catch this error by checking `if (popup is closed && could not establish connection)`, but checking if the popup is closed requires `chrome.runtime.getContexts()`

- This is a new api that [recently got proposed](https://github.com/w3c/webextensions/pull/358), and the only example I have found where it is used is [here](https://github.com/GoogleChrome/chrome-extensions-samples/tree/6f2e6164943d95f24116650052ba9d0dc6bb7fb3/functional-samples/sample.tabcapture-recorder)
- The example has `"minimum_chrome_version": "116"`, while the current stable version is 115, so I think we can leave it for now and catch it properly when 116 is out ![image](https://github.com/Theodo-UK/sustainability-calculator/assets/57725347/4fbeb7d2-b0e0-486e-a634-bd1eab0d92cb)